apps/site/assets/ts/__v3api.d.ts

```diff
@@ -17,6 +19,17 @@ export interface Headsign {
   train_number: string | null;
 }
 
+export interface HeadsignWithCrowding {
+  name: string;
+  time_data_with_crowding_list: PredictedOrScheduledTimeWithCrowding[];
+  train_number: string | null;
+}
```

Comment:
Is this supposed to have a headsign?

Reply:
Also wondering if you can get by by editing the existing Headsign interface:

```ts
export interface Headsign {
  name: string;
  headsign?: string;
  times: PredictedOrScheduledTime[] | PredictedOrScheduledTimeWithCrowding[];
  train_number: string | null;
}
```

Reply:
I'm not sure what `headsign` means in other areas that use `Headsign`, but in this case that property is never used. The `name` property is the headsign name. (Part of what I found confusing with the `Headsign` type—apart from the fact that it is not actually modeling a headsign—is that it was (is?) being used for multiple different collections of data coming from different types of server requests.)

Reply:
I didn't want to adapt the existing interface because `PredictedOrScheduledTime` and `PredictedOrScheduledTimeWithCrowding` aren't equivalent sorts of data. `PredictedOrScheduledTimeWithCrowding` has an additional level of objects. Plus `HeadsignWithCrowding` and `Headsign` aren't used in the same part of the app, so there was no benefit to them being closer in naming. Maybe they should be named more differently?

But also, there is a lot of room here to continue cleaning up this data and be more consistent about naming and what we are modeling. Really what this data represents (at least, I think) is a trip. The trip is on a route pattern which has a headsign, and the trip has a time it's scheduled to be at a certain stop and potentially there might be a prediction about when the vehicle running the trip might actually get to the stop. There might also be information about how crowded the vehicle running the trip currently is. I think naming things along those lines might make everything a lot clearer, but I was reluctant to change more than necessary on the client side in this PR.